### DISCUSSION TITLE
输入ip地址打不开admin

### DISCUSSION THREAD

**Discussion by evolumin** [WongSaang/chatgpt-ui#141] [Q&A] [answered]
你好，我按照一键部署的方式把程序部署在azure服务器上，但是用 ip:9000/admin的方式打不开网页，请问怎么解决🙏

**Comment by WongSaang** [comment]
打开时什么样的呢？

**Reply by evolumin** [reply]
Safari浏览器无法打开页面
Safar浏览器无法打开页面“52.146.41.123:9000/admin”，因为服务器意外中断了连接。这种情况在服务器忙时有时会出现。请等几分钟再试。

**Reply by WongSaang** [reply]
服务器的 9000 是否对外开放，检查安全组等。

**Comment by cgofun** [comment]
Forbidden (403)
CSRF verification failed. Request aborted.

More information is available with DEBUG=True.

:9000/admin 总是这个提示，进不去

**Reply by evolumin** [reply]
我也是这个问题，9000设置好以后，输入用户名admin，密码password，就出来这个：

“Forbidden (403)
CSRF verification failed. Request aborted.
More information is available with DEBUG=True.”

求助作者大大

**Reply by WongSaang** [reply]
输入的ip是服务器的公网ip 吗？
`123.123.123.123:9000`
是这种格式吗？重新执行下命令，检查有没有中文字符

**Reply by cgofun** [reply]
之前都是正常的，今天升级到最新的，登录不了了，配置的是域名，域名无法登录，IP也登录不了，清了浏览器cookie也不行

**Reply by WongSaang** [reply]
你的部署方式是什么？命令行么？

**Reply by cgofun** [reply]
命令行，docker拉取，之前也是一样部署的

**Reply by WongSaang** [reply]
我又试了，没有复现这个问题。可以提供你们的配置内容么？

**Reply by cgofun** [reply]
> 我又试了，没有复现这个问题。可以提供你们的配置内容么？

配置内容是指啥，重新拉取，所有配置都重置了。半小时前我又复制那句部署命令重新部署了下，9000/admin还是一样错误提示

**Reply by WongSaang** [reply]
我可能之前的描述有误，运行命令后，让输入域名或者IP 时，不需要输入端口号。看看这个地方输入对了吗？

**Reply by cgofun** [reply]
肯定是对的，之前已经部署过好几遍了。输入域名，端口全部默认，登录页面都是正常的，就是登录不了，提示那个错误，配置时用ip和域名都一样，这个应该是登录代码处理跨站请求引起的，DJango框架这个问题挺常见的

**Reply by tangxinyuGit** [reply]
在使用一键部署脚本在vps上部署后，如果直接使用IP登录管理后台，是正常的。如果使用域名登录，则会提示csrf forbidden

**Reply by WongSaang** [reply]
@tangxinyuGit 那请重新运行脚本，在第一步输入域名

**Reply by cgofun** [reply, marked as answer]
知道是怎么回事了，运行docker-compose up --pull always -d启动就会出现CSRF这个错误，部署脚本和配置都没有问题，只运行部署脚本部署后是正常登录的。

**Reply by WongSaang** [reply]
嗯呢，脚本没有直接修改 docker-compose.yml 文件。所以直接运行 docker-compose up --pull always -d 参数会丢失

**Reply by evolumin** [reply]
我是一键部署的请问接下来我要怎么操作才能正常登陆？🙏

**Reply by WongSaang** [reply]
现在怎么个不正常？请具体描述。

**Reply by evolumin** [reply]
您好，还是之前的问题，显示后台登录界面，但输入用户名密码后显示：
Forbidden (403)
CSRF verification failed. Request aborted.
More information is available with DEBUG=True.

**Reply by cgofun** [reply]
你的问题是用这句docker-compose up --pull always -d造成的，带上这个参数：APP_DOMAIN="172.16.11.247:9000"  docker-compose up --pull always -d 这样运行就正常了
